app/customer.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+import math
+from typing import List
+
+from app.shops import Shop
+from app.car import Car
+
+
+class Customer:
+    def __init__(self,
+                 name: str,
+                 product_cart: dict,
+                 location: list,
+                 money: int,
+                 car: Car) -> None:
+        self._name = name
+        self._product_cart = product_cart
+        self._location = location
+        self._money = money
+        self._car = car
+
+    def money(self) -> None:
+        print(f"{self._name} has {self._money} dollars")
+
+    def customer_in_shop(self, shops: List[Shop]) -> None:
+        cheaper_shop = None
+        cheaper_shop_cost = None
+        for shop in shops:
+            distance = math.sqrt((shop.location[0] - self._location[0]) ** 2
+                                 + (shop.location[1] - self._location[1]) ** 2)
+            price = round(self._car.fuel_consumption / 100
+                          * distance * 2 * 2.4
+                          + shop.buy_products(self._product_cart), 2)
+            if cheaper_shop_cost is not None and price < cheaper_shop_cost:
+                cheaper_shop = shop
+                cheaper_shop_cost = price
+            elif cheaper_shop_cost is None:
+                cheaper_shop = shop
+                cheaper_shop_cost = price
```

Comment:
better to make separated method for blocks like this, would be much more readable